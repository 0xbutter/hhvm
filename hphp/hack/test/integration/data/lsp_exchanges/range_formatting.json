[
    {
      "jsonrpc": "2.0",
      "id": 1,
      "method": "initialize",
      "params":
        {
          "initializationOptions":{},
          "processId":null,
          "rootPath":"${root_path}",
          "capabilities":{}
        }
    },
    {
      "jsonrpc": "2.0",
      "method": "$test/waitForHhServerReady",
      "params": {}
    },
    {
      "jsonrpc": "2.0",
      "method": "textDocument/didOpen",
      "params": {
         "textDocument": {
           "uri":"${php_file_uri}",
           "languageId": "hack",
           "version":1,
           "text":"${php_file}"
         }
       }
    },
    {
      "jsonrpc": "2.0",
      "method": "textDocument/rangeFormatting",
      "id": 2,
      "params": {
          "textDocument": {
            "uri":"${php_file_uri}"
          },
          "range": {
            "start": {
              "line": 4,
              "character": 0
            },
            "end": {
                "line": 5,
                "character": 0
            }
          },
          "options": {"tabSize": 5, "insertSpaces": true}
       }
    },
    {
      "jsonrpc": "2.0",
      "id": 3,
      "method": "shutdown",
      "params": {}
    }
    ]
