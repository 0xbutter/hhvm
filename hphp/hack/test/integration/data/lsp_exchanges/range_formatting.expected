[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "typeDefinitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": [
      {
        "range": {
          "start": {
            "line": 4,
            "character": 0
          },
          "end": {
            "line": 5,
            "character": 0
          }
        },
        "newText": "     $a = \"this\";\n"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": null
  }
]
